develop/blockchains/deployment/generate-chain-specs.md

```diff
@@ -1,4 +1,133 @@
+- Any other genesis state for a pallet
+
+Nodes also require the compiled Wasm to execute the runtime logic on the chain, so the initial runtime must also be supplied in the chain specification.
+
+## Declaring storage items for a runtime
+
+A runtime usually requires some storage items to be configured at genesis. This includes the initial state for pallets, for example, how much balance `Alice` or `Bob` has, or the account with sudo permissions.
+
+These storage values are configured in the genesis portion of the chain specification. You can either create a [patch](https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/index.html#chain-spec-formats){target=_blank}  file for [`chain-spec-builder`](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/chain_spec_genesis/index.html){target=_blank} to ingest, or if you're using `chain_spec.rs`, the pallet can be configured in the code.
+
+## Chain specification JSON format
+
+Users generally work with the JSON format of the chain specification. Internally, the chain specification is embodied by the [`GenericChainSpec`](https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/struct.GenericChainSpec.html){target=\_blank} struct, with specific properties accessible through the [`ChainSpec`](https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/trait.ChainSpec.html){target=\_blank} struct. The chain specification includes the following keys:
+
+- `name` - the human-readable name for the network
+- `id` - the machine-readable identifier for the network
+- `chainType` - the type of chain to start (refer to [ChainType](https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/enum.ChainType.html){target=\_blank} for more details)
+- `bootNodes` - a list of multiaddresses belonging to the chain's boot nodes
+- `telemetryEndpoints` - an optional list of multiaddresses for telemetry endpoints, with verbosity levels ranging from 0 to 9 (0 being the lowest verbosity)
+- `protocolId` - the optional protocol identifier for the network
+- `forkId` - an optional fork ID that should typically be left empty; it can be used to signal a fork at the network level when two chains share the same genesis hash
+- `properties` - custom properties provided as a key-value JSON object
+- `codeSubstitutes` - an optional mapping of block numbers to Wasm code
+- `genesis` - the genesis configuration for the chain
+
+For example, the following JSON shows a basic chain specification file:
+
+```json
+--8<-- "code/develop/blockchains/deployment/generate-chain-specs/basic-chain-specs.json"
+``` 
+
+
+## Creating a Custom Chain Specification
+
+To create a custom chain specification, you can use the [`chain-spec-builder`](https://paritytech.github.io/polkadot-sdk/master/staging_chain_spec_builder/index.html){target=\_blank} tool. This is a CLI tool that is used to generate chain specifications from the runtime of a node. To install the tool, run the following command:
```

Comment:
The phrase `following command` is used many times throughout the page. Replace in some places with synonyms